src/v/model/timeout_clock.h

```diff
@@ -27,4 +27,12 @@ static constexpr timeout_clock::time_point no_timeout
 static constexpr timeout_clock::duration max_duration
   = timeout_clock::duration::max();
 
+inline model::timeout_clock::time_point
+time_from_now(model::timeout_clock::duration d) {
+    /// Saturate now() + duration at no_timeout, avoid overflow
+    const auto now = model::timeout_clock::now();
+    const auto remaining = model::no_timeout - now;
```

Comment:
let's not assume the implementation of `no_timeout` is based on max.

Reply:
Ok point, but to me kinda seems odd to type out `model::timeout_clock::time_point::max()` when it is defined 5 lines above in this very file. Thoughts?

Reply:
it's like depending on `std::string::npos` having a value `-1` except we don't have a standardization process that guarantees how `no_timeout` is implemented.